cpp/Platform.Collections/Stacks/CStack.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+namespace Platform::Collections::Stacks
+{
+    template<typename TSelf, typename TElement>
+    concept CStack =
+        requires(TSelf & self, TElement item)
+    {
+        { self.push(item) };
+
+        { self.pop() };
+
+        { self.top() } -> std::same_as<TElement&>;
```

Comment:
Do we need check this method here? Isn’t it enough to just check the const version below?